sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> The info w.r.t Agent Upgrade. </summary>
+    public partial class AgentUpgrade
+    {
+        /// <summary> Initializes a new instance of AgentUpgrade. </summary>
+        public AgentUpgrade()
+        {
+        }
+
+        /// <summary> Initializes a new instance of AgentUpgrade. </summary>
+        /// <param name="desiredVersion"> Specifies the version info w.r.t AgentUpgrade for the machine. </param>
+        /// <param name="correlationId"> The correlation ID passed in from RSM per upgrade. </param>
+        /// <param name="enableAutomaticUpgrade"> Specifies if RSM should try to upgrade this machine. </param>
+        /// <param name="lastAttemptTimestamp"> Timestamp of last upgrade attempt. </param>
+        /// <param name="lastAttemptStatus"> Specifies the status of Agent Upgrade. </param>
+        /// <param name="lastAttemptMessage"> Failure message of last upgrade attempt if any. </param>
+        internal AgentUpgrade(string desiredVersion, string correlationId, bool? enableAutomaticUpgrade, string lastAttemptTimestamp, LastAttemptStatusEnum? lastAttemptStatus, string lastAttemptMessage)
+        {
+            DesiredVersion = desiredVersion;
+            CorrelationId = correlationId;
+            EnableAutomaticUpgrade = enableAutomaticUpgrade;
+            LastAttemptTimestamp = lastAttemptTimestamp;
+            LastAttemptStatus = lastAttemptStatus;
+            LastAttemptMessage = lastAttemptMessage;
+        }
+
+        /// <summary> Specifies the version info w.r.t AgentUpgrade for the machine. </summary>
+        public string DesiredVersion { get; set; }
+        /// <summary> The correlation ID passed in from RSM per upgrade. </summary>
+        public string CorrelationId { get; set; }
+        /// <summary> Specifies if RSM should try to upgrade this machine. </summary>
+        public bool? EnableAutomaticUpgrade { get; set; }
+        /// <summary> Timestamp of last upgrade attempt. </summary>
+        public string LastAttemptTimestamp { get; }
```

Comment:
Could this be a type of `DateTimeOffset`? See an example [here](https://github.com/Azure/azure-sdk-for-net/blob/e4fce1607d65d709b19c81cc78969cd08b804244/sdk/hdinsight/Azure.ResourceManager.HDInsight/src/autorest.md?plain=1#L146).

Reply:
changed the type of LastAttemptTimestamp to DateTimeOffset.